src/etc/gdb_load_rust_pretty_printers.py

```diff
@@ -1,3 +1,17 @@
+# Add this folder to the python sys path; GDB Python-interpreter will now find modules in this path
+import sys
+from os import path
+self_dir = path.dirname(path.realpath(__file__))
+sys.path.append(self_dir)
+
 import gdb
 import gdb_lookup
-gdb_lookup.register_printers(gdb.current_objfile())
+
+# current_objfile can be none; even with `gdb foo-app`; sourcing this file after gdb init now works
+try:
+    current_objfile = gdb.lookup_objfile(gdb.selected_inferior().progspace.filename)
```

Comment:
In gdb, a pretty-printer can be installed either on an objfile, a progspace, or globally.  So, you don't really need to try to find an objfile in the progspace, you can just install it there directly.  Or, you could install it globally, up to you; the difference only matters if you are debugging multiple inferiors and only want Rust pretty-printers (or different versions of Rust pretty-printers) in some of them.

See https://sourceware.org/gdb/current/onlinedocs/gdb.html/Selecting-Pretty_002dPrinters.html#Selecting-Pretty_002dPrinters

For comparison, libstdc++ ends up doing something like:

```
if obj is None:
   obj = gdb    # install globally
register_printers (obj)
```

Reply:
Thanks @tromey; admittedly I went with the first approach that mimicked the "current object file" (but without going through `gdb.current_objfile` as it seems to always be null, for simple applications loaded with `gdb fooapp`).

What do you think would be the best approach? And what approach would make users familiar with `rust-gdb`, witness the same behavior (and thus expect the same results) when doing
```
gdb fooapp
gdb> source path/to/pplib.py
```
? If that question makes any sense.

Reply:
> Thanks @tromey; admittedly I went with the first approach that mimicked the "current object file" (but without going through `gdb.current_objfile` as it seems to always be `None`, for simple applications loaded with `gdb fooapp`).

Yeah, the current objfile is only set momentarily while loading associated scripts.

> What do you think would be the best approach?

Personally I'd have it register the pretty-printers on the current progspace when the current objfile is None.

Reply:
I think you're right. I've updated with a new commit to make sure that the behavior doesn't change for users of `rust-gdb`.

Reply:
FWIW, this looks good to me.